pkg/sql/conn_executor_exec.go

```diff
@@ -713,6 +713,11 @@ func (ex *connExecutor) execStmtInOpenState(
 	p.cancelChecker.Reset(ctx)
 
 	p.autoCommit = canAutoCommit && !ex.server.cfg.TestingKnobs.DisableAutoCommitDuringExec
```

Comment:
nit: `&& ex.extraTxnState.numDDL == 0`

I think this deserves a bit more commentary to say that this is about suppressing the commit during statement execution, but not that preventing the end of the statement from running auto-commit.

Reply:
Updated this language to:

	// Auto-commit is disallowed during statement execution if we previously executed any DDL.
	// This is because it may potentially create jobs and do other operations rather than
	// a KV commit. Instead, the connection executor will auto-commit after this statement,
	// and carry out any extra operations needed for DDL.